src/adapters/rpc-server/rpc-server.ts

```diff
@@ -79,38 +74,31 @@ export async function createRpcServerComponent({
         logger.info(`[RPC] RPC Server listening on port ${rpcServerPort}`)
       })
 
-      await pubsub.subscribeToChannel(FRIENDSHIP_UPDATES_CHANNEL, friendshipUpdateHandler(getContext, logger))
+      await pubsub.subscribeToChannel(FRIENDSHIP_UPDATES_CHANNEL, friendshipUpdateHandler(subscribersContext, logger))
       await pubsub.subscribeToChannel(
         FRIEND_STATUS_UPDATES_CHANNEL,
-        friendConnectivityUpdateHandler(getContext, logger, db)
+        friendConnectivityUpdateHandler(subscribersContext, logger, db)
       )
     },
     attachUser({ transport, address }) {
       const normalizedAddress = normalizeAddress(address)
 
       transport.on('close', () => {
-        if (SHARED_CONTEXT.subscribers[normalizedAddress]) {
-          delete SHARED_CONTEXT.subscribers[normalizedAddress]
-          logger.debug('User disconnected and removed from subscribers', { address: normalizedAddress })
-        }
+        subscribersContext.removeSubscriber(normalizedAddress)
```

Comment:
What about moving the normalized address within the subscribers ' context so clients can abstract of doing that? Or do we also need them normalized for other workflows here?